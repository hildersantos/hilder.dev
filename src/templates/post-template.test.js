import React from "react"
import { render } from "react-testing-library"

import PostTemplate from "./post-template"
import fixtureSiteInfo from "../../tests/__fixtures__/fixture-site-info"
import fixtureSingle from "../../tests/__fixtures__/fixture-single"
import { useStaticQuery, StaticQuery } from "gatsby"

const props = {
  ...fixtureSiteInfo,
  ...fixtureSingle,
}

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => render(props)),
    useStaticQuery.mockReturnValue(props)
})

describe("PostTemplate", () => {
  it("renders without errors", () => {
    const { asFragment } = render(<PostTemplate data={props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders a title", () => {
    const { container, getByText } = render(<PostTemplate data={props} />)

    expect(container.querySelector("h1")).toBeInTheDocument()
    expect(
      getByText(fixtureSingle.markdownRemark.frontmatter.title)
    ).toBeInTheDocument()
  })
})
