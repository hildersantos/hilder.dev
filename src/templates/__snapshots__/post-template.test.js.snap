// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders without errors 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="container"
    >
      <header
        class="header-inner"
      >
        <a
          class="header-inner__link--home header-inner__link"
          href="/"
        >
          Test Title
        </a>
      </header>
      <article
        class="single"
      >
        <header
          class="single__header"
        >
          <h1
            class="single__title"
          >
            test
          </h1>
          <time
            class="single__date"
            datetime="2016-09-01T00:00:00-03:00"
          >
            01/09/2016
          </time>
        </header>
        <div
          class="single__content"
        >
          <p>
            test
          </p>
        </div>
      </article>
    </div>
  </main>
</DocumentFragment>
`;
